### PR TITLE
Guide: Intersphinx in Read the Docs

### PR DESCRIPTION
**Pull request by stsewd**
This guide is about how to use intersphinx in general and how to use it in rtd (subprojects, translations, etc) and also with private projects in the commercial site.

### CODE REVIEW

File: docs/guides/intersphinx.rst [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+       'sub-apis': ('https://docs.readthedocs.io/projects/apis/en/latest', None),
+   }
+
+Intersphinx with private projects
+---------------------------------
+
+If you are using :doc:`/commercial/index`,
+Intersphinx will not be able to fetch the inventory file from private docs.
+
+Intersphinx supports `URLs with Basic Authorization <https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html#using-intersphinx-with-inventory-file-under-basic-authorization>`__,
```
Comment (stsewd) [inline docs/guides/intersphinx.rst]: sphinx doesn't have a link to this section (actually I added that section :grin:) and they don't use the `autosectionlabel` extension :(

File: docs/guides/intersphinx.rst [hunk 2]
```diff
@@ -0,0 +1,143 @@
+Link to Other Project's Documentation With Intersphinx
+======================================================
+
+You may be familiar with using the :ref:`:ref: role <sphinx:ref-role>` to link to any location of your docs.
+It helps you to keep all links up to date, Sphinx will raise a warning if the reference doesn't exist.
+
+Some times you may need to link to a location of another documentation project.
+We could just link to where the documentation is hosted,
+and use Sphinx's ``linkcheck`` builder to check for broken links.
```
Comment (stsewd) [inline docs/guides/intersphinx.rst]: Sphinx doesn't have any example on their docs to link to, but this usually a target in their makefile `make linkcheck`

Comment (stsewd) [reply docs/guides/intersphinx.rst]: And not sure if is worth mentioning here, but I wanted to make clear that users can still have that feature from :ref: without using intersphinx

File: docs/guides/intersphinx.rst [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+   RTD_TOKEN_DOCS = os.environ.get('RTD_TOKEN_DOCS')
+
+   intersphinx_mapping = {
+       # Links to a private project named "docs"
+       'docs': (f'https://{RTD_TOKEN_DOCS}:@readthedocs-docs.readthedocs-hosted.com/en/latest', None),
+   }
+
+.. note::
+
+   Another way of using Intersphinx with private projects is to download the inventory file and keep it in sync when the project changes.
```
Comment (stsewd) [inline docs/guides/intersphinx.rst]: I'm about to put this as another paragraph and not inside a note, not sure

File: docs/guides/intersphinx.rst [hunk 4]
```diff
@@ -0,0 +1,143 @@
+Link to Other Project's Documentation With Intersphinx
+======================================================
+
+You may be familiar with using the :ref:`:ref: role <sphinx:ref-role>` to link to any location of your docs.
+It helps you to keep all links up to date, Sphinx will raise a warning if the reference doesn't exist.
```
[the remark this answers is not in the document]
Comment (stsewd) [reply docs/guides/intersphinx.rst]: Yeah, wasn't sure what else to put to convince a user of using intersphinx o explaining how it works without going too deeper, the ref role seemed like a simple example

**Comment by stsewd** [comment]
Thanks for the review @davidfischer! I have moved the mention of `linkcheck` to a note and also mention how to get the targets used in Intersphinx.

File: docs/guides/intersphinx.rst [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+   - :ref:`:ref: role <sphinx:ref-role>`
+   - :doc:`sphinx:usage/extensions/intersphinx`
+   - :doc:`Intersphinx <sphinx:usage/extensions/intersphinx>`
+
+Result:
+
+- :ref:`sphinx:ref-role`
+- :ref:`:ref: role <sphinx:ref-role>`
+- :doc:`sphinx:usage/extensions/intersphinx`
+- :doc:`Intersphinx <sphinx:usage/extensions/intersphinx>`
```
[the remark this answers is not in the document]
Comment (stsewd) [reply docs/guides/intersphinx.rst]: Done! I also wanted to add a recommendation of using the [autosectionlabel extension](https://www.sphinx-doc.org/en/master/usage/extensions/autosectionlabel.html) to make it easy to link to all headers without having to create targets manually, not sure if it fits in this guide.

File: docs/guides/intersphinx.rst [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+.. note::
+
+   You can also use Sphinx's ``linkcheck`` builder to check for broken links.
+   By default it will also check the validity of ``#anchors`` in links.
+
+   .. prompt:: bash $
+
+      sphinx-build -b linkcheck . _build/linkcheck
+
+   See all the `options for the linkcheck builder <https://www.sphinx-doc.org/en/master/usage/configuration.html#options-for-the-linkcheck-builder>`__.
```
Comment (davidfischer) [inline docs/guides/intersphinx.rst]: A perfect example of something that can't be linked to with Intersphinx.

File: docs/guides/intersphinx.rst [hunk 7]
```diff
@@ -0,0 +1,163 @@
+Link to Other Projects' Documentation With Intersphinx
+======================================================
+
+You may be familiar with using the :ref:`:ref: role <sphinx:ref-role>` to link to any location of your docs.
+It helps you to keep all links within your docs up to date and warns you if a reference target moves or changes
+so you can ensure that your docs don't have broken cross-references.
+
+Some times you may need to link to a location of another documentation project.
+We could just link to where the documentation is hosted.
+
```
Comment (davidfischer) [inline docs/guides/intersphinx.rst]: I would remove the small paragraph above and combine it into the paragraph below. Something like:

```
Sometimes you may need to link to a specific section of another project's documentation.
While you could just hyperlink directly, there is a better way.
```

**Comment by stsewd** [comment]
Failure on rtd is unrelated.

**Pull request merged**